Core/Core/Router/CourseTabUrlInteractor.swift

```diff
@@ -0,0 +1,173 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import Combine
+
+public final class CourseTabUrlInteractor {
+
+    private struct TabModel {
+        let id: String
+        let htmlUrl: String
+    }
+
+    private var enabledTabsPerCourse: [Context: [String]] = [:]
+    private var subscriptions = Set<AnyCancellable>()
+
+    public init() { }
+
+    public func setupTabSubscription() {
+        let useCase = LocalUseCase<Tab>(scope: .all)
+        return ReactiveStore(useCase: useCase)
+            .getEntitiesFromDatabase(keepObservingDatabaseChanges: true)
+            .replaceError(with: [])
+            .sink { [weak self] tabs in
+                self?.tabsDidUpdate(tabs)
+            }
+            .store(in: &subscriptions)
```

Comment:
We are storing this subscription that never ends `keepObservingDatabaseChanges: true` so we should cancel this when the user logs out not to keep the old coredata context stack alive.